sockeye/model.py

```diff
@@ -198,12 +201,15 @@ def _get_embed_weights(self, prefix: str) -> Tuple[mx.sym.Symbol, mx.sym.Symbol,
         :param prefix: Prefix.
         :return: Tuple of source and target parameter symbols.
         """
-        w_embed_source = mx.sym.Variable(prefix + C.SOURCE_EMBEDDING_PREFIX + "weight",
+        w_embed_source = None
```

Comment:
I think you can undo the change here. This is just creating the variable but its never used in the case of PassThroughEmbedding, so you don't need to if isinstance here.

Reply:
Ok. I will run the tests to check that it is working properly.

Reply:
I just saw that the attribute `self.config.config_embed_source.vocab_size` needs to be defined, which does not exist for image captioning. So I would leave it as it is in my pull request, if you do not have other alternatives.

Reply:
Sorry, not sure I get this. Could you elaborate a little bit why this is related?

Reply:
The attribute `vocab_size` does not exist because there is no vocabulary for the source since our source is images.

Reply:
ah ok, because of the shape definition of that variable.